packages/sdk-client-v3/src/middleware/create-concurrent-modification-middleware.ts

```diff
@@ -5,7 +5,13 @@ import {
   Next,
 } from '../types/types'
 
-export default function createConcurrentModificationMiddleware(): Middleware {
+export default function createConcurrentModificationMiddleware(
+  customLogic?: (
+    version: number,
+    request: MiddlewareRequest,
+    response: MiddlewareResponse
+  ) => Promise<string>
```

Comment:
This return type need not he a string alone, it should be the Typeof body which I believe is either a string, Buffer or an object.